File: VladislavShalnev/Term 2/Block 4/Task 8 (Bash)/Bash/Commands/CatCommand.cs
namespace Bash.Commands;

public class CatCommand : ICommand
{
	public string Trigger => "cat";
	
	public string[] Run(string[] args)
	{
		var result = new List<string>();

		foreach (var arg in args)
			result.Add(File.ReadAllText(arg));

		return result.ToArray();
	}
}

File: IvanShurenkov/Second Term/Block4/Bash/lib/bash/src/main/kotlin/lib/pipe/commands/CatCommand.kt
package lib.pipe.commands

import lib.pipe.Output
import java.io.File

class CatCommand(val name: String) : ICommand {
    override fun run(args: Array<String>, input: String?): Output {
        val output = Output(null)
        for (arg in args) {
            try {
                val text = File(arg).readText()
                output.addOut("$text")
            } catch (e: Exception) {
                //print(e.message)
                output.addErr(e.message)
            }
        }
        return output
    }
}